llvm/lib/Transforms/Utils/CloneFunction.cpp

```diff
@@ -268,6 +268,8 @@ void llvm::CloneFunctionInto(Function *NewFunc, const Function *OldFunc,
 
   // Loop over all of the instructions in the new function, fixing up operand
   // references as we go. This uses VMap to do all the hard work.
+  assert(!OldFunc->empty() && "Cannot clone module with empty function body.\
```

Comment:
what if the function is really empty?

How about use OldFunc->assertModuleIsMaterialized() instead?

Reply:
Good catch. I tested and even empty functions are OK as long as they're materialized. I'll update this check